File: Recursion_3.py
#!/bin/python3
# Complete the factorial function below.
def factorial(n):
    if(n==1):
        return 1
    else:
        result = n*factorial(n-1)
    return result

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    result = factorial(n)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Perl/Recursion_3.pl
#!/usr/bin/perl

use strict;
use warnings;
open(my $fptr, '>', $ENV{'OUTPUT_PATH'});

my $n = <STDIN>;
chomp $n;
my $temp=0;
my $result = factorial($n);
print $fptr "$result\n";
close $fptr;

sub factorial 
{
my $a=pop(@_);
    if($a==1)
    {
        return $a;
    }
    else
    {
        $temp = factorial($a-1)*$a; 
    }
return $temp;

}